File: Caesar_Cipher/caeser.java
import java.util.Scanner;

public class caeser {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter 0 to encrypt and 1 to decrypt: ");
        int encdec = scanner.nextInt();

        if (encdec == 0) {
            System.out.print("Enter plain text to encrypt: ");
            String plainText = scanner.next();
            System.out.println(caeser.encrypt(plainText));
            scanner.close();
        } else if (encdec == 1) {
            System.out.print("Enter cypher text to decrypt: ");
            String cipherText = scanner.next();
            System.out.println(caeser.decrypt(cipherText));
            scanner.close();
        } else {
            System.out.println("Enter either 0 or 1");
            scanner.close();
        }
    }

    public static String encrypt(String plainText) {
        String cipher_text = "";
        char[] characters = plainText.toCharArray();

        for (char character : characters) {
            int newCharacterAscii = ((int) character + 3) % 127;

            if (newCharacterAscii < 32) {
                char newCharacter = (char) (32 + newCharacterAscii);
                cipher_text += newCharacter;
            } else {
                cipher_text += (char) newCharacterAscii;
            }
        }
        return cipher_text;
    }

    public static String decrypt(String cipherText) {
        String plain_text = "";
        char[] characters = cipherText.toCharArray();

        for (char character : characters) {
            int originalCharacterAscii = ((int) character - 3) % 127;

            if (originalCharacterAscii < 32) {
                char originalCharacter = (char) (127 - (32 - originalCharacterAscii));
                plain_text += originalCharacter;
            } else {
                plain_text += (char) originalCharacterAscii;
            }
        }

        return plain_text;
    }
}

File: Caesar_Cipher/caeser.js
const readLine = require('readline').createInterface({
	input: process.stdin,
	output: process.stdout
});

function encrypt(plainText) {
	let cipherText = '';
	[...plainText].forEach(character => {
		const newCharacterAscii = (character.charCodeAt() + 3) % 127;

		if (newCharacterAscii < 32) {
			const newCharacter = String.fromCharCode(32 + newCharacterAscii);
			cipherText += newCharacter;
		} else {
			const newCharacter = String.fromCharCode(newCharacterAscii);
			cipherText += newCharacter;
		}
	});

	return cipherText;
}

function decrypt(cipherText) {
	let plainText = '';
	[...cipherText].forEach(character => {
		const originalCharacterAscii = (character.charCodeAt() - 3) % 127;

		if (originalCharacterAscii < 32) {
			const originalCharacter = String.fromCharCode(
				127 - (32 - originalCharacterAscii)
			);
			plainText += originalCharacter;
		} else {
			const originalCharacter = String.fromCharCode(originalCharacterAscii);
			plainText += originalCharacter;
		}
	});

	return plainText;
}

readLine.question('Enter 0 to encrypt or 1 to decrypt: ', answer => {
	if (answer === '0') {
		readLine.question('Enter plain text to encrypt: ', plainText => {
			const cipherText = encrypt(plainText);
			console.log(cipherText);
			readLine.close();
		});
	} else if (answer === '1') {
		readLine.question('Enter cipher text to decrypt: ', cipherText => {
			const plainText = decrypt(cipherText);
			console.log(plainText);
			readLine.close();
		});
	} else {
		console.log('Enter either 0 or 1');
		readLine.close();
	}
});
